relay-server/src/actors/project_buffer.rs

```diff
@@ -296,6 +296,16 @@ impl BufferService {
 
                 // spawn the task to enqueue the entire buffer
                 tokio::spawn(async move {
+                    // A RESERVED lock will be acquired when the first INSERT, UPDATE, or DELETE statement is executed.
+                    // Once we started spooling our in-memory buffer, the dequeue from the disk
+                    // will be blocked till this operations either fails, or commits.
+                    if let Err(err) = sqlx::query("BEGIN").execute(&db).await {
+                        relay_log::error!("failed to initiate a transaction: {}", LogError(&err));
```

Comment:
Shouldn't you exit early if begin fails? I'd refactor this to an async function that returns an error and then catch it in the spawn to log errors.